src/Chainweb/Miner/Coordinator.hs

```diff
@@ -94,7 +114,7 @@ newtype PrimedWork =
 -- the current block's payload.
 --
 newtype MiningState = MiningState
-    (M.Map (T2 BlockHash BlockPayloadHash) (T3 Miner PayloadWithOutputs (Time Micros)))
+    (M.Map BlockPayloadHash (T3 Miner PayloadData (Time Micros)))
```

Comment:
It is not clear to me why the parent block hash is used in the key. I am not sure if the parent header (or some other unique value like the block height or creation time) actually gets included in the payload hash (probably not). But even if the payload isn't unique for a block (which may, unfortunately, be the case), we wouldn't care here.

(Also, using it as first component slows down the comparison, because many different payloads share the same parent hash, e.g. each miner has one payload hash for each parent. If we are concerned about memory cache performance in case of very large maps, we should use the block height instead, but I don't think that's an issue at this point.)

Reply:
We should think about whether "nodes supporting multiple miners" is a feature we want to continue to support. In this case I support moving to whatever the lookup needs, which this looks good for.

Reply:
As a general note, I don't prefer to use `T3` in a core datastructure. It's one thing to use tuples in processing code but in this case a proper type with a constructor and fields would be self-documenting.